src/shillelagh/adapters/base.py

```diff
@@ -51,6 +51,11 @@ class Adapter:
     # if true, the requested columns will be passed to ``get_rows`` and ``get_data``
     supports_requested_columns = False
 
+    # if true, a set from an IN statement will be assigned the new IN operator and
+    # passed in as a tuple to bounds in ``get_rows`` and ``get_data``. This requires
+    # supports_requested_columns to be set to true as well.
+    supports_in_statements = False
```

Comment:
I'm not sure if we need explicit support at the adapter level. I think we might be able to just add the `IN` adapter, and if a given column doesn't support it we could replace it with a bunch of `Equal` operators.